src/main/java/io/confluent/idesidecar/restapi/kafkarest/RecordSerializer.java

```diff
@@ -50,35 +48,38 @@ public ByteString serialize(
       return null;
     }
 
-    var serdeConfigs = clientConfigurator.getSerdeConfigs(schema, isKey);
-    if (schema.isEmpty()) {
-      return serializeSchemalessData(topicName, serdeConfigs, data, isKey);
-    }
-    var jsonNode = objectMapper.valueToTree(data);
-    var parsedSchema = schema.get().parsedSchema();
-    return switch (SchemaFormat.fromSchemaType(parsedSchema.schemaType())) {
-      case AVRO -> serializeAvro(
-          client, parsedSchema, serdeConfigs, topicName, jsonNode, isKey);
-      case JSON ->
-          serializeJsonSchema(client, parsedSchema, serdeConfigs, topicName, jsonNode, isKey);
-      case PROTOBUF ->
-          serializeProtobuf(client, parsedSchema, serdeConfigs, topicName, jsonNode, isKey);
-    };
-  }
+    try {
+      var serdeConfigs = clientConfigurator.getSerdeConfigs(schema, isKey);
+      if (schema.isEmpty()) {
+        return serializeSchemalessData(topicName, serdeConfigs, data, isKey);
+      }
+      var jsonNode = objectMapper.valueToTree(data);
 
-  private ByteString serializeAvro(
-      SchemaRegistryClient client,
-      ParsedSchema parsedSchema,
-      Map<String, String> configs,
-      String topicName,
-      JsonNode data,
-      boolean isKey
-  ) {
-    try (var avroSerializer = new KafkaAvroSerializer(client)) {
-      avroSerializer.configure(configs, isKey);
-      var schema = (AvroSchema) parsedSchema;
-      var record = wrappedToObject(() -> AvroSchemaUtils.toObject(data, schema));
-      return ByteString.copyFrom(avroSerializer.serialize(topicName, record));
+      // If we were sent bytes, we should just use them as-is.
+      if (ByteArrayJsonUtil.smellsLikeBytes(jsonNode)) {
+        return ByteString.copyFrom(ByteArrayJsonUtil.asBytes(jsonNode));
+      }
```

Comment:
This is the other functional change introduced in this PR -- though undocumented, if the key/value data is a JSON containing the base64 encoded string version of their raw bytes, like `{"__raw__": "AABdfvdsf=="}`, then we simply convert it into a `ByteString` and return it. There's no need to "serialize" it since it would have already been given to us serialized (if using schema).